lib/commands/deploy/index.js

```diff
@@ -105,25 +113,15 @@ class DeployCommand extends AbstractCommand {
  * @param {Function} callback
  */
 function deployResources(options, callback) {
-    const { profile, doDebug, ignoreHash } = options;
-    // Skill Metadata
-    Messenger.getInstance().info('==================== Deploy Skill Metadata ====================');
-    const skillMetaSpinner = new SpinnerView();
-    skillMetaSpinner.start('Uploading the entire skill package (it may take few minutes to build the skill metadata)...');
-    helper.deploySkillMetadata(options, (metaErr) => {
-        skillMetaSpinner.terminate();
-        if (metaErr && metaErr !== 'The hash of current skill package folder does not change compared to the last deploy hash result, '
-            + 'CLI will skip the deploy of skill package.') {
-            return callback(metaErr);
+    const { profile, doDebug, target, ignoreHash } = options;
+    _deploySkillMetadata(options, (err) => {
+        if (err) {
+            return callback(err);
         }
-        if (metaErr) {
-            // this case is the warning message of the same hash skip, deploy will continue
-            Messenger.getInstance().warn(metaErr);
-        } else {
-            ResourcesConfig.getInstance().write();
-            Messenger.getInstance().info('Skill package deployed successfully.');
+
+        if (target === CONSTANTS.DEPLOY_TARGET.SKILL_METADATA) {
+            return callback();
         }
```

Comment:
if skill is not created with an ID, we should not deploy skill infrastrucutre.

Reply:
do you mean checking for existence of skill id?

ResourcesConfig.getInstance().getSkillId(profile)

Reply:
Yes, skillId must have been existing before the following logic executes

Reply:
We will need to also gracefully let users know that they need to have a valid skillId for the proj first here

Reply:
actually just tested. it would true for the opposite case too. if we try to just deploy skill metadata before having an endpoint we get.
[Error]: {
  "message": "Skill is not ready for enablement.",
  "violations": [
    {
      "code": "MISSING_REQUIRED_PROPERTY",
      "message": "Object instance at property path \"$.manifest.apis.custom\" has missing required property: \"endpoint\".",
      "validationDetails": {
        "originalInstance": {
          "propertyPath": "$.manifest.apis.custom",
          "type": "BODY"
        },
        "requiredProperty": "endpoint"
      }
    }
  ]
}